clients/keycloak/src/main/kotlin/DefaultKeycloakClient.kt

```diff
@@ -383,6 +383,51 @@ class DefaultKeycloakClient(
         }.getOrElse {
             throw KeycloakClientException("Failed to load client roles for user '${id.value}'.", it)
         }.body()
+
+    override suspend fun addUserToGroup(username: UserName, groupName: GroupName) {
+        // FIXME: Unfortunately, no way to distinguish if there was a technical problem or the user was not found.
+        val user = try { getUser(username) } catch (e: KeycloakClientException) {
+            throw UserNotFoundException("User with username '${username.value}' not found.", e)
+        }
+        val group = getGroup(groupName)
```

Comment:
What if a non-existing group name is specified, would this throw a Keycloak exception as well? So, does this require some error handling, too?

Reply:
It was not a good idea to introduce this UserNotFound exception and try to fix the shortcomings of the AuthorizationService and KeycloakService APIs, because they don't make a difference between business exceptions and technical exceptions. I actually don't want to fix these interfaces within this feature task, so whenever something goes wrong, a KeycloakClientException will be thrown and propagated through the layers.